code/modules/mob/mob_helpers.dm

```diff
@@ -14,6 +14,49 @@
 	if(isStunned() || restrained())
 		return 1
 
+/mob/proc/get_screen_colour()
+	if(!client)
+		return 0
+	if(M_NOIR in mutations)
```

Comment:
Nuh uh it doesn't fire every player tick (unless you've drank detcoffee but I can't find a way to do chems otherwise) I specifically made it fire only on login/out or when the mob is told to update the colour.